psl-core/src/main/java/org/linqs/psl/util/MathUtils.java

```diff
@@ -155,4 +155,83 @@ public static BigInteger factorial(BigInteger number) {
 
         return result;
     }
+
+    /**
+     * Given a vector in cartesian coordinates,
+     * return the vector in a log scale.
+     */
+    public static double[] toLogScale(double[] vector, double logBase) {
+        double[] logScaleVector = new double[vector.length];
+
+        for (int i = 0; i < vector.length; i++) {
+            logScaleVector[i] = Math.pow(logBase, vector[i]);
+        }
+
+        return logScaleVector;
+    }
+
```

Comment:
Extra newline.

Reply:
Not sure where the extra newline is and how to correct this.

Reply:
There are two newlines (172 and 173), but there should only be one.